src/main/java/com/commercetools/sync/products/templates/beforeupdatecallback/SyncSingleLocale.java

```diff
@@ -136,42 +139,106 @@
                                                                             UpdateAction<Product>> updateActionMapper) {
         final LocalizedString newLocalizedField = newLocalizedFieldMapper.apply(newDraft);
         final LocalizedString oldLocalizedField = oldLocalizedFieldMapper.apply(oldProduct.getMasterData().getStaged());
-        if (oldLocalizedField == null && newLocalizedField == null) {
-            return Optional.empty();
+        if (oldLocalizedField != null && newLocalizedField != null) {
+            // if both old and new localized fields are set, only update if the locale values are not equal.
+            final String newLocaleValue = newLocalizedField.get(locale);
+            final String oldLocaleValue = oldLocalizedField.get(locale);
+
+            // At this point, we are sure both old locale value and new locale value have different values.
```

Comment:
how we are sure?

Reply:
because this method should only be called on the beforeUpdate -> which means that they are already different. Because if they are the same, the callback won't be called at all.